src/components/screens/chat/chatFormBlocks/ChatFormInputs.js

```diff
@@ -131,35 +134,61 @@ export default function ChatFormInputs({
 
   // vv  Attachments pick  vv //
   const pickUserFiles = () => filePicker.current.click();
-  const handlerChange = (event) => {
+  const handlerChange = async (event) => {
     if (!event.target.files.length) {
       return;
     }
+    setIsLoadingFile(true);
 
     const selectedFiles = [];
-    for (const file of event.target.files) {
-      if (file.name.length > 255) {
-        showCustomAlert(
-          "The file name should not exceed 255 characters.",
-          "warning"
-        );
-        return;
-      }
-      if (!file.type.startsWith("image/")) {
-        showCustomAlert("Please select an image file.", "warning");
-        return;
-      }
+    try {
+      for (const file of event.target.files) {
+        if (file.name.length > 255) {
+          throw new Error("The file name should not exceed 255 characters.", {
+            cause: {
+              message: "The file name should not exceed 255 characters.",
+            },
+          });
+        }
 
-      selectedFiles.push(file);
-    }
+        if (!file.type.startsWith("image/") && !file.name.includes(".HEIC")) {
+          throw new Error("Please select an image file.", {
+            cause: { message: "Please select an image file." },
+          });
+        } else if (file.name.includes(".HEIC")) {
+          const blob = new Blob([file], { type: "image/heic" });
+
+          const pngBuffer = await heic2any({
+            blob,
+            toType: "image/png",
+          });
+
+          const pngBlob = new Blob([pngBuffer], { type: "image/png" });
+          const pngFile = new File([pngBlob], file.name, {
+            type: "image/png",
+          });
```

Comment:
I guess, to follow SOLID principles you can export this part of your code to 'utils' and then import here to perform a check:

```
const blob = new Blob([file], { type: "image/heic" });

          const pngBuffer = await heic2any({
            blob,
            toType: "image/png",
          });

          const pngBlob = new Blob([pngBuffer], { type: "image/png" });
          const pngFile = new File([pngBlob], file.name, {
            type: "image/png",
          });
```

Reply:
100% agree